manim/mobject/mobject.py

```diff
@@ -200,6 +200,13 @@ def construct(self):
                         c.animate(rate_func=there_and_back).shift(RIGHT),
                     )
 
+        .. warning::
+
+            ``.animate``
+             will interpolate the :class:`~.Mobject` between its points prior to ``.animate`` and its points after applying ``.animate`` to it. This may result in unexpected behavior when attempting to interpolate along paths, or rotations.
+             That means:  .animate doesn't know that it should do a nice rotation, it only interpolates between the old positions and the new positions of the points.
+             If you want animations to consider the points between, consider using :class:`~.ValueTracker` with updaters instead.
```

Comment:
Do the docs make clear elsewhere what this would mean?

Reply:
Not really, but a good example would be: `square.animate.rotate(PI)` which demonstrates the unexpected behavior

Reply:
I was referring more to the bit about using ValueTrackers instead, but that's a good point too.

Reply:
Here would be an example that solves the rotation with ValueTracker:
https://docs.manim.community/en/stable/examples.html#movingangle
Thats a good idea @jsonvillanueva . I can also make an example to illustrate both approaches, to see the difference.
Further, I think we could add a tutorial section "Tips and Tricks" where we can show some of these "Manim Do's and Don'ts. Also, the example regarding TextMobject would fit there: https://github.com/ManimCommunity/manim/pull/1269

Reply:
I'll come back to that.